### PR TITLE
fix(faker): Use `MarshalText` for faker timestamps

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/3627.

**The TLDR** is that we should `MarshalText` when faking time data, as that follows an RFC.

**The DON'T READ part is**:
If we don't do it and default to `String()` that sometimes creates a string that cannot be parsed by `const defaultStringFormat = "2006-01-02 15:04:05.999999999 -0700 MST"`.

You can see an example in https://github.com/cloudquery/cloudquery/issues/3627 that the `2110-10-19 08:53:50.21670207 +0000 UTC m=+2775422040.494306874` string was created.
The reason that the string can't be parsed is because the nano seconds part has 8 digits and not 9, that is `21670207` instead of `216702070`.
The nano seconds string needs to be padded only when adding the `m=+2775422040.494306874` part. This was added in go 1.19 (see https://pkg.go.dev/time#Time.String).
To summarize:
1. `time.Parse("2006-01-02 15:04:05.999999999 -0700 MST", "2110-10-19 08:53:50.21670207 +0000 UTC")` works
2. `time.Parse("2006-01-02 15:04:05.999999999 -0700 MST", "2110-10-19 08:53:50.216702070 +0000 UTC  m=+2775422040.494306874")` works
3. `time.Parse("2006-01-02 15:04:05.999999999 -0700 MST", "2110-10-19 08:53:50.21670207 +0000 UTC  m=+2775422040.494306874")` fails

So technically this could be a bug in go, that needs to pad the nano seconds when adding the `m=+` part, though there is no guarantee on the format of `time.String()`.

I added another test `TestTimestamptzSetString` that shows this behavior. The only way I could find to create a string that has the `m=+` part is using `time.Now()`. It's in the PR to demonstrate the issue and I'll remove it before merging

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: schema/timestamptz_test.go [hunk 1]
```diff
@@ -21,19 +27,23 @@ func TestTimestamptzSet(t *testing.T) {
...
-		}
+		require.NoError(t, err)
+		require.Equal(t, tt.result, r)
+	}
+}
 
-		if !r.Equal(&tt.result) {
-			t.Errorf("%d: %v != %v", i, r, tt.result)
-		}
+func TestTimestamptzSetString(t *testing.T) {
```
Comment (erezrokah) [inline schema/timestamptz_test.go:43]: I'll remove this before merging the PR. This is here just to show the bug

**Comment by erezrokah** [comment]
>The reason that the string can't be parsed is because the nano seconds part has 8 digits and not 9, that is 21670207 instead of 216702070.

This is only partially correct. I found other use cases the parsing fails (even with 9 digits).

File: schema/timestamptz_test.go [hunk 2]
```diff
@@ -21,19 +27,27 @@ func TestTimestamptzSet(t *testing.T) {
...
+	}
+}
 
-		if !r.Equal(&tt.result) {
-			t.Errorf("%d: %v != %v", i, r, tt.result)
+func TestTimestamptzSetString(t *testing.T) {
+	for i := 0; i < 100000; i++ {
+		timeString := time.Now().Add(time.Duration(rand.Int63())).String()
+		_, err := time.Parse(defaultStringFormat, timeString)
+		if err == nil {
```
Comment (erezrokah) [inline schema/timestamptz_test.go:47]: Both errors are printed

**Review by hermanschaaf — APPROVED** [review]
Amazing work! 👏

**Comment by erezrokah** [comment]
> So technically this could be a bug in go, that needs to pad the nano seconds when adding the `m=+` part, though there is no guarantee on the format of `time.String()`.

I posted a message about this in https://groups.google.com/g/golang-nuts (I think it needs to be approved first)

**Review by yevgenypats — APPROVED** [review]
